### PR TITLE
Feature integration specs for views n+1

### PR DESCRIPTION
**Pull request by Alibaba2023**
# Changes in this PR:

## Team members: @Alibaba2023 and @hajnaloltyan
- [ ] Fixed the N+1 problem
- Before:
##
Screenshot 2023-11-07 004417
##
Screenshot 2023-11-07 004456

- After:
Screenshot 2023-11-07 010335
##
Screenshot 2023-11-07 010412
##
- [ ] Integration test added for posts and users

### CODE REVIEW

File: spec/features/posts_index.spec.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      expect(page).to have_text(post.text)
+    end
+  end
+
+  it 'shows the number of comments and likes' do
+    visit user_posts_path(users(:one))
+
+    expect(page).to have_text("Comments: #{posts(:one).comments.count}")
+    expect(page).to have_text("Likes: #{posts(:one).likes.count}")
+  end
```
Comment (Ibizugbe) [inline spec/features/posts_index.spec.rb]: Nice job so far as your project has captured almost all the test cases😉. but just as required, kindly add a test that asserts you can see the first comment on a post😉.
image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Alibaba2023 👋🏼👋🏼👋🏼, 

request change


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] All written tests are passing👍🏼.
- [x] A majority of the test cases has been captured👍🏼.
- [x] Capybara gen is present in the gemfile👍🏼.


## Required Changes ♻️

- [ ] Kindly adjust the naming convention for your test files `post_index_spec.rb` , `post_show_spec.rb`, `user_index_spec.rb` NOT `post_index.spec.rb` etc. as you have used. The test files not properly named will be ignored when you run the test😉.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/posts_index.spec.rb (hunk 1 above)
Comment (Alibaba2023) [reply spec/features/posts_index.spec.rb]: Thank you for the feedback.

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @Alibaba2023 , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done adding the remaining test. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**